packages/kbn-i18n/src/messages.ts

```diff
@@ -27,7 +27,9 @@ export interface Messages {
 }
 
 export interface PlainMessages {
```

Comment:
__issue:__ while you're here, let's finally improve this code and get rid of this weird `PlainMessages` interface and instead of having `Messages` and `PlainMessages` interfaces, let's have one:
```ts
export interface Translation {
  /**
   * Actual translated messages.
   */
  messages: Record<string, string>;

  /**
   * Locale of the translated messages.
   */
  locale: string;

  /**
   * Set of options to the underlying formatter.
   */
  formats?: Formats;
}
```

And in the rest of the code, let's use `translation` where `Translation` is used and `messages` where `Translation['messages']` is used (aka `i18n.getMessages().messages` looks wrong, it should probably be `i18n.getTranslation(s)().messages` or something like this). I think both `locale` and `messages` are non-optional? If so, let's mark them as such.